### PR TITLE
reference: update documents about pessimistic transaction

### PR DESCRIPTION
**Pull request by ran-huang**
### What is changed, added or deleted? 

Update documents about pessimistic transactions.

### What is the related PR or file link(s)? 

- Reference link(s):
- This PR is to align with: https://github.com/pingcap/docs-cn/pull/2263, https://github.com/pingcap/docs-cn/pull/2315
- [ ] N/A (not applicable)

**Important Notice:**

**If your changes apply to multiple TiDB documentation versions**, to trigger the bot to cherry-pick this PR to other release branches, make sure you add labels such as:

- **needs-cherry-pick-3.0**
- **needs-cherry-pick-3.1**
- **needs-cherry-pick-2.1**

### CODE REVIEW

**Comment by ran-huang** [comment]
Resolved conflict with https://github.com/pingcap/docs/pull/1875.

**Comment by ran-huang** [comment]
This PR does not pass CI check, because it has a link pointing to a new document in https://github.com/pingcap/docs-cn/pull/2198 that is still in translation. 
So this PR will proceed after the translation is completed.

**Comment by TomShawn** [comment]
> This PR does not pass CI check, because it has a link pointing to a new document in [pingcap/docs-cn#2198](https://github.com/pingcap/docs-cn/pull/2198) that is still in translation.
> So this PR will proceed after the translation is completed.

#2198 is now merged. You can proceed with this PR now.

**Comment by ran-huang** [comment]
> > This PR does not pass CI check, because it has a link pointing to a new document in [pingcap/docs-cn#2198](https://github.com/pingcap/docs-cn/pull/2198) that is still in translation.
> > So this PR will proceed after the translation is completed.
> 
> #2198 is now merged. You can proceed with this PR now.

#2198 is merged but its aligned PR in docs https://github.com/pingcap/docs/pull/1695 is still in progress. I have to wait until the `optimistic-transaction.md` file in #1695 is merged.

**Comment by TomShawn** [comment]
> #2198 is merged but its aligned PR in docs #1695 is still in progress. I have to wait until the `optimistic-transaction.md` file in #1695 is merged.

OK

**Comment by lilin90** [comment]
@ran-huang I see that @anotherrachel closed #1695.

**Comment by ran-huang** [comment]
> @ran-huang I see that @anotherrachel closed #1695. PTAL~

Thanks.

**Comment by ran-huang** [comment]
CI fails due to a dealink. Waiting for  https://github.com/pingcap/docs/pull/2088

**Comment by ran-huang** [comment]
@lilin90 @youjiali1995 PTAL~

File: reference/transactions/transaction-pessimistic.md [hunk 1]
```diff
@@ -6,9 +6,31 @@ category: reference
...
+
+To apply the pessimistic transaction mode, choose any of the following three methods that suits your needs:
+
+- Execute the `BEGIN PESSIMISTIC;` statement to allow the transaction to apply the pessimistic transaction mode. You can write it in comment style as `BEGIN /*!90000 PESSIMISTIC */;` to make it compatible with the MySQL syntax.
+
+- Execute the `set @@tidb_txn_mode = 'pessimistic';` statement to allow all the explicit transactions (namely non-autocommit transactions) processed in this session to apply the pessimistic transaction mode.
+
+- Execute the `set @@global.tidb_txn_mode = 'pessimistic';` statement to allow all newly created sessions of the entire cluster to apply the pessimistic transaction mode to execute explicit transactions.
+
+After you set `global.tidb_txn_mode` to `pessimistic`, the pessimistic transaction mode is applied by default. To apply the optimistic transaction mode to the transaction, you can use either of the following three methods:
```
Comment (lilin90) [inline reference/transactions/transaction-pessimistic.md:21]: `either` usually means one of two items.
```suggestion
After you set `global.tidb_txn_mode` to `pessimistic`, the pessimistic transaction mode is applied by default. To apply the optimistic transaction mode to the transaction, you can use any of the following three methods:
```

File: reference/transactions/transaction-pessimistic.md [hunk 2]
```diff
@@ -22,40 +44,48 @@ Pessimistic transactions in TiDB behave similarly to those in MySQL. See the min
...
+    When a write conflict occurs, the optimistic transaction is terminated directly, but the pessimistic transaction retries the statement with the latest data until there is no write conflict. The log prints this entry with each retry, so there is no need for extra attention.
+
+2. When DML is executed, an error `pessimistic lock retry limit reached` is returned.
+
+    In the pessimistic transaction mode, every statement has a retry limit. This error is returned when the retry times of write conflict exceeds the limit. The default retry limit is `256`. To change the limit, modify the `max-retry-limit` under the `[pessimistic-txn]` category in the TiDB configuration file.
+
+3. The execution time limit for pessimistic transactions.
 
-    When `SELECT LOCK IN SHARE MODE` is specified in a statement, it has the same effect as that without the lock, so the read or write of other transactions is not blocked.
+    The execution time of transactions cannot exceed the limit of `tikv_gc_life_time`. Apart from that, the pessimistic transactions have a TTL (Time to Live) limit of 10 minutes, so the pessimistic transactions that execute over 10 minutes might fail to commit.
```
Comment (lilin90) [inline reference/transactions/transaction-pessimistic.md:91]:
```suggestion
    The execution time of transactions cannot exceed the limit of `tikv_gc_life_time`. In addition, the pessimistic transactions have a TTL (Time to Live) limit of 10 minutes, so the pessimistic transactions that execute over 10 minutes might fail to commit.
```

**Pull request merged**